<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LSystemUi</class>
 <widget class="QMainWindow" name="LSystemUi">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>902</width>
    <height>589</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>lsystem - An interactive simulator for Lindenmayer systems</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="topMainLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>761</width>
      <height>223</height>
     </rect>
    </property>
    <property name="maximumSize">
     <size>
      <width>16777215</width>
      <height>242</height>
     </size>
    </property>
    <layout class="QHBoxLayout" name="topMainLayout">
     <item>
      <widget class="QFrame" name="wdgSettings">
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>221</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>8</number>
        </property>
        <property name="topMargin">
         <number>8</number>
        </property>
        <property name="rightMargin">
         <number>8</number>
        </property>
        <property name="bottomMargin">
         <number>8</number>
        </property>
        <item>
         <widget class="QWidget" name="wdgDefinitions" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <property name="spacing">
            <number>8</number>
           </property>
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QWidget" name="widget_2" native="true">
             <property name="minimumSize">
              <size>
               <width>85</width>
               <height>0</height>
              </size>
             </property>
             <widget class="QLabel" name="labelDefinitions_2">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>0</y>
                <width>79</width>
                <height>51</height>
               </rect>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>0</width>
                <height>50</height>
               </size>
              </property>
              <property name="layoutDirection">
               <enum>Qt::LeftToRight</enum>
              </property>
              <property name="text">
               <string>Definitions:</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
              </property>
             </widget>
             <widget class="QPushButton" name="cmdAdd">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>20</y>
                <width>40</width>
                <height>25</height>
               </rect>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>19</width>
                <height>25</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>40</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="toolTip">
               <string>Add a literal at the end of the list</string>
              </property>
              <property name="text">
               <string>add</string>
              </property>
             </widget>
             <widget class="QLabel" name="label">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>74</y>
                <width>79</width>
                <height>50</height>
               </rect>
              </property>
              <property name="text">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;Start&lt;br/&gt;symbol:&lt;/body&gt;&lt;/html&gt;</string>
              </property>
             </widget>
             <widget class="QLabel" name="lblStartSymbol">
              <property name="geometry">
               <rect>
                <x>65</x>
                <y>92</y>
                <width>25</width>
                <height>31</height>
               </rect>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="styleSheet">
               <string notr="true">background-color: rgb(211, 215, 207);
border: 1px solid black;
margin: 2px 5px 2px 0px;</string>
              </property>
              <property name="text">
               <string>A</string>
              </property>
             </widget>
             <widget class="QPushButton" name="cmdRemove">
              <property name="geometry">
               <rect>
                <x>44</x>
                <y>20</y>
                <width>40</width>
                <height>25</height>
               </rect>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>19</width>
                <height>25</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>40</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="toolTip">
               <string>Add a literal at the end of the list</string>
              </property>
              <property name="text">
               <string>del</string>
              </property>
             </widget>
            </widget>
           </item>
           <item>
            <widget class="QTableView" name="tblDefinitions">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="toolTip">
              <string>Edit definitions. Use drag&amp;drop for moving them.</string>
             </property>
             <property name="dragEnabled">
              <bool>true</bool>
             </property>
             <property name="dragDropOverwriteMode">
              <bool>false</bool>
             </property>
             <property name="dragDropMode">
              <enum>QAbstractItemView::InternalMove</enum>
             </property>
             <property name="defaultDropAction">
              <enum>Qt::MoveAction</enum>
             </property>
             <property name="selectionMode">
              <enum>QAbstractItemView::SingleSelection</enum>
             </property>
             <property name="selectionBehavior">
              <enum>QAbstractItemView::SelectItems</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="wdgAdditionalSettings" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>72</height>
           </size>
          </property>
          <widget class="FocusableLineEdit" name="txtIter" native="true">
           <property name="geometry">
            <rect>
             <x>310</x>
             <y>33</y>
             <width>51</width>
             <height>21</height>
            </rect>
           </property>
           <property name="toolTip">
            <string>The recursion depth</string>
           </property>
           <property name="styleSheet">
            <string notr="true">background-color: rgb(255, 255, 255);</string>
           </property>
           <property name="text" stdset="0">
            <string>3</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_4">
           <property name="geometry">
            <rect>
             <x>2</x>
             <y>6</y>
             <width>71</width>
             <height>20</height>
            </rect>
           </property>
           <property name="text">
            <string>Turn (°):</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_6">
           <property name="geometry">
            <rect>
             <x>230</x>
             <y>6</y>
             <width>61</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>right (-):</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_9">
           <property name="geometry">
            <rect>
             <x>230</x>
             <y>31</y>
             <width>81</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Iterations:</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_8">
           <property name="geometry">
            <rect>
             <x>3</x>
             <y>31</y>
             <width>121</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Step size (px):</string>
           </property>
          </widget>
          <widget class="FocusableLineEdit" name="txtLeft" native="true">
           <property name="geometry">
            <rect>
             <x>150</x>
             <y>7</y>
             <width>51</width>
             <height>21</height>
            </rect>
           </property>
           <property name="toolTip">
            <string>The left turn (mathematically positive) for each occurrence of &quot;+&quot;</string>
           </property>
           <property name="styleSheet">
            <string notr="true">background-color: rgb(255, 255, 255);</string>
           </property>
           <property name="text" stdset="0">
            <string>30</string>
           </property>
          </widget>
          <widget class="FocusableLineEdit" name="txtRight" native="true">
           <property name="geometry">
            <rect>
             <x>310</x>
             <y>7</y>
             <width>51</width>
             <height>21</height>
            </rect>
           </property>
           <property name="toolTip">
            <string>The right turn (mathematically negative) for each occurrence of &quot;-&quot;</string>
           </property>
           <property name="styleSheet">
            <string notr="true">background-color: rgb(255, 255, 255);</string>
           </property>
           <property name="text" stdset="0">
            <string>-30</string>
           </property>
          </widget>
          <widget class="FocusableLineEdit" name="txtStep" native="true">
           <property name="geometry">
            <rect>
             <x>150</x>
             <y>33</y>
             <width>51</width>
             <height>21</height>
            </rect>
           </property>
           <property name="toolTip">
            <string>The length of each painted line segment</string>
           </property>
           <property name="styleSheet">
            <string notr="true">background-color: rgb(255, 255, 255);</string>
           </property>
           <property name="text" stdset="0">
            <string>30</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_2">
           <property name="geometry">
            <rect>
             <x>95</x>
             <y>6</y>
             <width>51</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>left (+):</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_7">
           <property name="geometry">
            <rect>
             <x>379</x>
             <y>31</y>
             <width>65</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>[] Scaling:</string>
           </property>
          </widget>
          <widget class="FocusableLineEdit" name="txtScaleDown" native="true">
           <property name="geometry">
            <rect>
             <x>470</x>
             <y>33</y>
             <width>51</width>
             <height>21</height>
            </rect>
           </property>
           <property name="toolTip">
            <string>The step size is multiplied by this factor for each occurrence of &quot;[&quot;</string>
           </property>
           <property name="styleSheet">
            <string notr="true">background-color: rgb(255, 255, 255);</string>
           </property>
           <property name="text" stdset="0">
            <string>0.8</string>
           </property>
          </widget>
          <widget class="QLabel" name="label_10">
           <property name="geometry">
            <rect>
             <x>380</x>
             <y>5</y>
             <width>81</width>
             <height>21</height>
            </rect>
           </property>
           <property name="text">
            <string>Start angle:</string>
           </property>
          </widget>
          <widget class="FocusableLineEdit" name="txtStartAngle" native="true">
           <property name="geometry">
            <rect>
             <x>470</x>
             <y>7</y>
             <width>51</width>
             <height>21</height>
            </rect>
           </property>
           <property name="toolTip">
            <string>The initial angle, 0° points right, -90° point upwards</string>
           </property>
           <property name="styleSheet">
            <string notr="true">background-color: rgb(255, 255, 255);</string>
           </property>
           <property name="text" stdset="0">
            <string>-90</string>
           </property>
          </widget>
          <widget class="QCheckBox" name="chkAutoPaint">
           <property name="geometry">
            <rect>
             <x>3</x>
             <y>52</y>
             <width>231</width>
             <height>23</height>
            </rect>
           </property>
           <property name="text">
            <string>Auto paint (interactive mode)</string>
           </property>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="frmPresets">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>230</width>
         <height>200</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>221</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <widget class="QLabel" name="label_5">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>10</y>
          <width>67</width>
          <height>17</height>
         </rect>
        </property>
        <property name="text">
         <string>Configs:</string>
        </property>
       </widget>
       <widget class="QPushButton" name="cmdStore">
        <property name="geometry">
         <rect>
          <x>80</x>
          <y>190</y>
          <width>61</width>
          <height>25</height>
         </rect>
        </property>
        <property name="toolTip">
         <string/>
        </property>
        <property name="text">
         <string>store</string>
        </property>
       </widget>
       <widget class="QPushButton" name="cmdDelete">
        <property name="geometry">
         <rect>
          <x>150</x>
          <y>190</y>
          <width>71</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>delete</string>
        </property>
       </widget>
       <widget class="QPushButton" name="cmdLoad">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>190</y>
          <width>61</width>
          <height>25</height>
         </rect>
        </property>
        <property name="text">
         <string>load</string>
        </property>
       </widget>
       <widget class="QListView" name="lstConfigs">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>211</width>
          <height>151</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>Doubleclick to load a config</string>
        </property>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layPaintFrameWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>240</y>
      <width>761</width>
      <height>151</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="layPaint">
     <item>
      <widget class="QFrame" name="frmPaint">
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QVBoxLayout" name="layPaintFrame">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <property name="spacing">
           <number>0</number>
          </property>
          <item>
           <widget class="QWidget" name="wdgOut" native="true">
            <property name="cursor">
             <cursorShape>CrossCursor</cursorShape>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="footerWidget" native="true">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>0</height>
        </size>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="ClickableLabel" name="lblStatus" native="true">
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(211, 215, 207);
border: 1px solid black;</string>
          </property>
          <property name="text" stdset="0">
           <string/>
          </property>
          <property name="margin" stdset="0">
           <number>3</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="cmdSettings">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Settings</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="cmdAbout">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>About</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>902</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ClickableLabel</class>
   <extends>QWidget</extends>
   <header>util/clickablelabel.h</header>
  </customwidget>
  <customwidget>
   <class>FocusableLineEdit</class>
   <extends>QWidget</extends>
   <header>util/focusablelineedit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
